scripts/zkai_attention_kv_stwo_typed_size_estimate_gate.py

```
@@ -0,0 +1,611 @@
+def build_rows() -> list[dict[str, Any]]:
+    specs = artifact_specs(section_delta_rows())
+    cli = run_stwo_size_estimate([spec["path"] for spec in specs])
+    if cli.get("schema") != "zkai-stwo-proof-size-estimate-cli-v1":
+        raise StwoTypedSizeEstimateGateError("CLI schema drift")
+    if cli.get("accounting_source") != ACCOUNTING_SOURCE:
+        raise StwoTypedSizeEstimateGateError("CLI accounting source drift")
+    if cli.get("stable_binary_serializer_status") != STABLE_BINARY_SERIALIZER_STATUS:
+        raise StwoTypedSizeEstimateGateError("CLI binary serializer status drift")
+    cli_rows = cli.get("rows")
+    if not isinstance(cli_rows, list) or len(cli_rows) != len(specs):
+        raise StwoTypedSizeEstimateGateError("CLI row count drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)